src/SortTrait.php

```diff
@@ -13,32 +13,41 @@
 {
     /**
      * Sorts the elements in the entire Collection<T> using the specified comparer.
+     *
      * @param ComparerInterface $comparer The ComparerInterface implementation to use when comparing elements, or null
-     * to use the default comparer Comparer<T>.Default.
+     *                                    to use the default comparer Comparer<T>.Default.
+     *
      * @return $this
      */
     public function sort(ComparerInterface $comparer = null)
     {
+        /** @var Dictionary $this */
         if ($comparer === null) {
             $comparer = $this->getDefaultComparer();
         }
-        usort($this->container, array($comparer, 'compare'));
+        $container = $this->getAll();
+        usort($container, [$comparer, 'compare']);
```

Comment:
The best way to go here would be to do something like this:

```
usort($$this, [$comparer, 'compare']);
```

As **$this** is going to be an *Iterable* it's going to be possible to sort it. Would you mind to test it?